vehicle/toyota/identity.go

```diff
@@ -0,0 +1,183 @@
+package toyota
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/evcc-io/evcc/util"
+	"github.com/evcc-io/evcc/util/request"
+	"github.com/golang-jwt/jwt/v5"
+	"golang.org/x/oauth2"
+)
+
+const (
+	APIVersion   = "protocol=1.0,resource=2.1"
+	ClientID     = "oneapp"
+	ClientSecret = "6GKIax7fGT5yPHuNmWNVOc4q5POBw1WRSW39ubRA8WPBmQ7MOxhm75EsmKMKENem"
+	Scope        = "openid profile vehicles"
+	Realm        = "a-ncb-prod"
+	RedirectURI  = "com.toyota.oneapp:/oauth2Callback"
+)
+
+type Identity struct {
+	log *util.Logger
+	*request.Helper
+	oauth2.TokenSource
+	uuid string
+}
+
+func NewIdentity(log *util.Logger) *Identity {
+	return &Identity{
+		log:    log,
+		Helper: request.NewHelper(log),
+	}
+}
+
+func (v *Identity) authenticate(auth Auth, user, password string, passwordSet bool) (*Token, error) {
+	uri := fmt.Sprintf("%s/%s", BaseUrl, AuthenticationPath)
+
+	// Update callbacks with credentials
+	for id, cb := range auth.Callbacks {
+		switch cb.Type {
+		case "NameCallback":
+			outputValue, ok := cb.Output[0].Value.(string)
+			if ok && outputValue == "User Name" {
+				auth.Callbacks[id].Input[0].Value = user
+			}
+		case "PasswordCallback":
+			auth.Callbacks[id].Input[0].Value = password
+			passwordSet = true
+		}
+	}
+
+	// Send authentication request
+	data, err := json.Marshal(auth)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal auth data: %w", err)
+	}
+	req, err := request.New(http.MethodPost, uri, bytes.NewReader(data), map[string]string{
+		"Content-type": "application/json",
+		"Accept":       "application/json",
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// If we've already set the password, expect a token response
+	if passwordSet {
+		var token Token
+		if err = v.DoJSON(req, &token); err != nil {
```

Comment:
```suggestion
		if err := v.DoJSON(req, &token); err != nil {
```

Reply:
I'm reassigning err created in line 40 instead of creating a new variable. Not a good way? (I'm pretty new to golang)

Reply:
It's correct but since you're never using the outside var might as well make it local for clarity.